kotlin-inject-runtime/src/main/kotlin/me/tatarka/inject/internal/LazyGet.kt

```diff
@@ -0,0 +1,16 @@
+package me.tatarka.inject.internal
+
+import java.util.concurrent.ConcurrentMap
+
+fun <T> ConcurrentMap<String, Any?>.lazyGet(key: String, init: () -> T): T {
```

Comment:
This is unfortunately java-specific, will have to figure out other platforms when that is possible. js is easy since there's no multithreading, not sure what to do about native.